cli/popper/gha.py

```diff
@@ -175,7 +176,11 @@ def check_secrets(self):
         for _, a in self.wf['action'].items():
             for s in a.get('secrets', []):
                 if s not in os.environ:
-                    pu.fail('Secret {} not defined\n.'.format(s))
+                    if os.environ.get("CI") == "true":
+                        pu.fail('Secret {} not defined\n.'.format(s))
+                    else:
```

Comment:
when `--dry-run` is given (`dry_run=True`) the prompt should be skipped as well, as users don't want to actually run the pipeline

Reply:
In the case of `dry-run` instead of letting the action fail, I have set the value for an undefined secret as an empty string. This should keep the action from failing and the user won't have to enter the value for the secret. 
Works?

Reply:
I think that would work. Although the empty value is not read after that. An alternative is to do:

```python
if self.dry_run:
    return
```

at the beginning of the `check_secrets` function